python/pyspark/sql/streaming/listener.py

```diff
@@ -62,6 +63,19 @@ class StreamingQueryListener(ABC):
     >>> spark.streams.addListener(MyListener())
     """
 
+    def _set_spark_session(self, spark):
+        self._sparkSession = spark
+
+    @property
+    def spark(self) -> Optional["SparkSession"]:
+        if hasattr(self, "_sparkSession"):
+            return self._sparkSession
+        else:
+            return None
+
+    def _init_listener_id(self):
```

Comment:
This is used for `removeListener`. We initialize a listener id with each listener, when removing, only send the id so the server side recognizes it (server side has a id-listener mapping cache) and removes it